### PR TITLE
Bugfix/zcs 5035

### PR DESCRIPTION
**Pull request by sneha-patil-synacor**
Added support for following scenarios:
1. If email is changed in sendCode, user should be able to send verification email. If it is not changed, the sendCode will not send the code again.
2. recovery email address should not same as primary address in sendCode.

Updated soap doc.
Updated junit tests.

### CODE REVIEW

File: store/src/java/com/zimbra/cs/service/mail/SetRecoveryEmail.java [hunk 1]
```diff
@@ -231,8 +248,9 @@ protected void reset(Mailbox mbox, ZimbraSoapContext zsc) throws ServiceExceptio
     }
 
     private static void validateEmail(String email, Account account) throws ServiceException {
-        String[] addresses = account.getMailAddress();
-        if (Arrays.asList(addresses).contains(email)) {
+        String accountName = account.getName();
+        String aliases[] = account.getAliases();
+        if (Arrays.asList(aliases).contains(email) || accountName.equals(email)) {
             throw ServiceException
                 .FAILURE("Recovery address should not be same as primary email address.", null);
```
Comment (swapnilpingle89) [inline store/src/java/com/zimbra/cs/service/mail/SetRecoveryEmail.java]: How about having failure message as "Recovery address should not be same as primary/aliases email address."?
User might enter alias email and argue it is not same as primary email.

File: store/src/java/com/zimbra/cs/service/mail/SetRecoveryEmail.java [hunk 2]
```diff
@@ -231,10 +248,11 @@ protected void reset(Mailbox mbox, ZimbraSoapContext zsc) throws ServiceExceptio
     }
 
     private static void validateEmail(String email, Account account) throws ServiceException {
-        String[] addresses = account.getMailAddress();
-        if (Arrays.asList(addresses).contains(email)) {
+        String accountName = account.getName();
+        String aliases[] = account.getAliases();
+        if (Arrays.asList(aliases).contains(email) || accountName.equals(email)) {
             throw ServiceException
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/service/mail/SetRecoveryEmail.java]: Do you want to log at the debug level what the 2 email addresses were.

**Comment by piyush4360** [comment]
For point 2, recovery email address not be same as primary email address, 

Are we verifying the aliases also for that user?

**Comment by sneha-patil-synacor** [comment]
@piyush4360 Yes, we are.

**Comment by sneha-patil-synacor** [comment]
Added logging. Added new logger called zimbra.passwordreset.

File: store/src/java/com/zimbra/cs/service/mail/SetRecoveryEmail.java (hunk 2 above)
Comment (sneha-patil-synacor) [reply store/src/java/com/zimbra/cs/service/mail/SetRecoveryEmail.java]: Done

**Pull request closed without merging**